File: 프로그래머스/1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.java
class Solution {
        public int[] solution(int n, int m) {
        int[] answer = {gcd(n,m),lcm(n,m)};
        return answer;
    }
    
    public static int gcd(int a, int b) {
    	if(b==0) {
    		return a;
    	}
    	return gcd(b,a%b);
    }
    public static int lcm(int a, int b) {
    	return a / gcd(a,b)*b;
    }
}

File: 프로그래머스/1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.kt
class Solution {
    fun solution(n: Int, m: Int): IntArray {
        var answer = intArrayOf()
        answer += gcd(n,m)
        answer += lcm(n,m)
        return answer
    }

    fun gcd(n: Int, m: Int): Int {
        return if (m != 0) gcd(m, n % m) else n
    }

    fun lcm(n: Int, m: Int): Int {
        return n * m / gcd(n, m)
    }
}